kubeflow/common/iap-ingress/base/config-map.yaml

```
@@ -181,6 +179,92 @@ data:
       echo "Backend updated successfully. Waiting 1 hour before updating again."
       sleep 3600
     done
+  setup_cloudendpoints.sh: |
+    #!/bin/bash
+    #
+    # A simple shell script to configure a cloud endpoint
+    set -x
+    [ -z ${NAMESPACE} ] && echo Error NAMESPACE must be set && exit 1
+    [ -z ${SERVICE} ] && echo Error SERVICE must be set && exit 1
+    [ -z ${INGRESS_NAME} ] && echo Error INGRESS_NAME must be set && exit 1
+    [ -z ${ENDPOINT_NAME} ] && echo Error ENDPOINT_NAME must be set && exit 1
+
+    __dir="$(cd "$(dirname "${BASH_SOURCE[0]}")" && pwd)"
+
+    PROJECT=$(curl -s -H "Metadata-Flavor: Google" http://metadata.google.internal/computeMetadata/v1/project/project-id)
+    if [ -z ${PROJECT} ]; then
+      echo Error unable to fetch PROJECT from compute metadata
+      exit 1
+    fi
+
+    PROJECT_NUM=$(curl -s -H "Metadata-Flavor: Google" http://metadata.google.internal/computeMetadata/v1/project/numeric-project-id)
+    if [ -z ${PROJECT_NUM} ]; then
+      echo Error unable to fetch PROJECT_NUM from compute metadata
+      exit 1
+    fi
+
+    # Activate the service account
+    if [ ! -z "${GOOGLE_APPLICATION_CREDENTIALS}" ]; then
+      # As of 0.7.0 we should be using workload identity and never setting GOOGLE_APPLICATION_CREDENTIALS.
+      # But we kept this for backwards compatibility but can remove later.
+      gcloud auth activate-service-account --key-file=${GOOGLE_APPLICATION_CREDENTIALS}
+    fi
+
+    # Print out the config for debugging
+    gcloud config list
+    gcloud auth list
+
+    set_endpoint () {
```

Should we exit early if it is already setup ?

```suggestion
    set_endpoint () {
        gcloud endpoints services describe ${ENDPOINT_NAME}
        if [ $? == 0 ] ; then
            echo "${ENDPOINT_NAME} cloud endpoint already setup"
            return 0
        fi
```

What about scenarios where we want to update it ?

---

The purpose of **cloud-endpoints-enabler** is to create and activate a cloud endpoint during deployment. It's supposed to be deleted at the end of `make apply` run. We do not support *update* scenario for now. @zijianjoy mentioned above an idea about having boolean flags to decide when to delete **cloud-endpoints-enabler**, **iap-enabler**, and **backend-updater** workloads.

---

Ok, got it.

We don't use the makefiles in our setup. We keep the manifests/kustomizations in git and use Fluxcd to update our fleet.
I need to think about how we can better handle the *enabler workloads...

Thanks for the clarification.